contracts/PrizeStrategyUpkeep.sol

```diff
@@ -70,25 +85,44 @@ contract PrizeStrategyUpkeep is KeeperCompatibleInterface, Ownable {
     /// @param performData Not used in this implementation.
     function performUpkeep(bytes calldata performData) override external {
 
+        uint256 _upkeepLastUpkeepBlockNumber = upkeepLastUpkeepBlockNumber;
+        require(block.number > _upkeepLastUpkeepBlockNumber + upkeepMinimumBlockInterval, "PrizeStrategyUpkeep::minimum block interval not reached");
+
         address[] memory prizePools = prizePoolRegistry.getAddresses();
 
+      
         uint256 batchCounter = upkeepBatchSize; //counter for batch
+
         uint256 poolIndex = 0;
-        
+        uint8 startAwardCounter = 0;
+        uint8 completeAwardCounter = 0;
+
+        uint256 updatedUpkeepBlockNumber;
+
         while(batchCounter > 0 && poolIndex < prizePools.length){
             
             address prizeStrategy = PrizePoolInterface(prizePools[poolIndex]).prizeStrategy();
             
             if(prizeStrategy.canStartAward()){
                 PeriodicPrizeStrategyInterface(prizeStrategy).startAward();
+                updatedUpkeepBlockNumber = block.number;
+                startAwardCounter++;
                 batchCounter--;
             }
             else if(prizeStrategy.canCompleteAward()){
                 PeriodicPrizeStrategyInterface(prizeStrategy).completeAward();
+                updatedUpkeepBlockNumber = block.number;
```

Comment:
I'm thinking you can do this just once outside of the loop by checking completeAwardCounter > 0 or startAwardCounter > 0